core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -236,8 +237,9 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         logManager.startup()
 
         metadataCache = new MetadataCache(config.brokerId)
-        tokenCache = new DelegationTokenCache(config.saslEnabledMechanisms)
-        credentialProvider = new CredentialProvider(config.saslEnabledMechanisms, tokenCache)
+        // Enable delegation token cache for all SCRAM mechanisms to simplify dynamic update
+        tokenCache = new DelegationTokenCache(ScramMechanism.mechanismNames)
```

Comment:
I'm not sure I understand why it makes sense to refer to the scram mechanisms here in place of `saslEnabledMechanisms`.

Reply:
We watch ZooKeeper for changes anyway, so rather than create the cache only if a mechanism is enabled, the code currently updates the cache anyway. Delegation tokens uses SCRAM, hence it is for all SCRAM mechanisms.